### PR TITLE
ci: Use CircleCI 2.0 (Migrate linux CI from CircleCI 1.0, Add Windows CI)

### PR DESCRIPTION
**Pull request by jcfr**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            name: python-3.6.4-x64
+            python_version: 3.6.4
+            python_arch: x64
+
+        - build-test-python:
+            name: python-3.6.4-x86
+            python_version: 3.6.4
+            python_arch: x86
+
+        # 3.5.5
```
Comment (thewtex) [inline .circleci/config.yml:42]: Note: 3.5 is no longer supported by CPython upstream.

**Review by thewtex — APPROVED** [review]
:+1:

**Comment by jcfr** [comment]
@fbordignon  This PR (started a while back) has been revisited to updates the CI based on your work done in https://github.com/python-cmake-buildsystem/python-cmake-buildsystem/pull/282

Once this is integrated, I will start moving forward with adding support for building Python 3.7, 3.8 and 3.9 by integrating your work.

In the meantime, would appreciate if you could review :pray:

**Comment by jcfr** [comment]
In the interest of time, I went ahead and integrated. This will allow me to get started on integrating the remaining of the work related the PR adding support for Python 3.9.x

**Comment by fbordignon** [comment]
Great, I am away this week. I can help next week.

**Comment by jcfr** [comment]
I completely understand. And to re-iterate, thanks again for all the work you did :pray: 

 Since python 3.6.x end of life was few days ago and I have some "quiet" time, I am moving forward with :rocket:

**Pull request merged**